### ISSUE TITLE
webpack 5 cannot store large cache

### ISSUE THREAD

**Issue by zlk89** [webpack/webpack#12126]
# Bug report

**What is the current behavior?**
Webpack is showing this warning:
`[webpack.cache.PackFileCacheStrategy] Caching failed for pack: RangeError [ERR_OUT_OF_RANGE]: The value of "value" is out of range. It must be >= -2147483648 and <= 2147483647. Received 7_244_357_265`

more loggings:
```
<t> [webpack.cache.PackFileCacheStrategy] resolve build dependencies: 9007.401757 ms
<t> [webpack.cache.PackFileCacheStrategy] snapshot build dependencies: 384.107231 ms
    [webpack.cache.PackFileCacheStrategy] Captured build dependencies
    [webpack.cache.PackFileCacheStrategy] 82083 fresh items in cache
<t> [webpack.cache.PackFileCacheStrategy] store pack: 65582.725819 ms
<w> [webpack.cache.PackFileCacheStrategy] Caching failed for pack: RangeError [ERR_OUT_OF_RANGE]: The value of "value" is out of range. It must be >= -2147483648 and <= 2147483647. Received 7_236_815_654
    [webpack.cache.PackFileCacheStrategy] RangeError [ERR_OUT_OF_RANGE]: The value of "value" is out of range. It must be >= -2147483648 and <= 2147483647. Received 7_236_815_654
        at checkInt (internal/buffer.js:62:11)
        at writeU_Int32LE (internal/buffer.js:682:3)
        at Buffer.writeInt32LE (internal/buffer.js:851:10)
        at serialize (xx/node_modules/webpack/lib/serialization/FileMiddleware.js:147:10)
        at async Promise.all (index 3)
        at serialize (xx/node_modules/webpack/lib/serialization/FileMiddleware.js:115:3)
```

**If the current behavior is a bug, please provide the steps to reproduce.**
- load a huge app

A potential similar issue: https://github.com/webpack/webpack/issues/11908 and steps: https://github.com/kelunik/webpack-5-testing/commit/a800896c6c10df3572d50b573ef03c9e2a8607f6

**What is the expected behavior?**
It should handle large file, or split the large file into smaller files.

**Other relevant information:**
webpack version:  5.9.0
Node.js version: v12.16.1
Operating System: MacOS
Additional tools:

**Comment by alexander-akait** [comment]
@zlk89 Please provide real steps to reproduce

**Comment by sokra** [comment]
Yep I know the problem

**Comment by sokra** [comment]
ok this should be fixed in [5.10.2](https://github.com/webpack/webpack/releases/tag/v5.10.2).

Could to test and verify that? Maybe even provide a report for your experience with the persistent cache. That would be interesting to hear 😄
